js/index.js

```diff
@@ -42,42 +57,119 @@ function renderPepperoni() {
   });
 }
 
+
+
+// Iteration 1: set the visibility of `<section class="mushroom">`
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function (oneMushroom) {
+    if (state.mushrooms === true) {
+      oneMushroom.style.visibility = 'visible';
+    } else {
+      oneMushroom.style.visibility = 'hidden';
+    }
+  });
+
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function (onePepper) {
+    if (state.greenPeppers) {
+      onePepper.style.visibility = 'visible';
+    } else {
+      onePepper.style.visibility = 'hidden';
+    }
+  })
+
+
+
 }
 
+
+
+
+
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(function (sauce) {
+    if (state.whiteSauce === false) {
+      sauce.removeAttribute("sauce-white");
+    } else {
+      sauce.setAttribute("class", "sauce-white");
+    }
+  })
+
+
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+
+  document.querySelectorAll('.crust').forEach(function (crust) {
+    if (state.glutenFreeCrust === false) {
+      crust.removeAttribute("crust-gluten-free");
+    } else {
+      sauce.setAttribute("class", "crust-gluten-free");
+    }
+  })
+
+
+
+
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+
+  document.querySelectorAll('.btn').forEach(function (btn) {
+    btn.removeAttribute("active");
+  })
 }
 
+
+
+
+document.querySelector('.btn.btn-pepperoni').addEventListener('click', () => {
+  console.log(random)
+  state.pepperoni = !state.pepperoni;
+  renderEverything();
+});
+
+
+document.querySelector('.btn-mushrooms').addEventListener('click', () => {
+  state.mushrooms = !state.mushrooms;
+  renderEverything();
+});
+
+
+document.querySelector('.btn-green-peppers').addEventListener('click', () => {
+  state.greenPeppers = !state.greenPeppers;
+  renderEverything();
+});
+
+document.querySelector('.btn-sauce').addEventListener('click', () => {
+  state.whiteSauce = !state.whiteSauce;
+  console.log("souce-test")
+  renderEverything();
+});
+
+
+
+
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+
 }
```

Comment:
For this function you can take a look at Jan solution:

function renderPrice() {
// Iteration 4: change the HTML of <aside class="panel price">
let price = basePrice;
let list = '';
for (let ingredient in state) {
if (state[ingredient]) {
list += <li>$ ${ingredients[ingredient].price} ${ingredients[ingredient].name}</li>;
price += ingredients[ingredient].price;
}
}
document.querySelector('.panel.price strong').innerText = $${price};
document.querySelector('.panel.price ul').innerHTML = list;
}